test/Riok.Mapperly.Tests/Mapping/EnumerableDeepCloningTest.cs

```diff
@@ -206,7 +206,22 @@ public void EnumerableOfPrimitivesToEnumerableDeepCloning()
     public void ReadOnlyCollectionOfPrimitivesToEnumerableDeepCloning()
     {
         var source = TestSourceBuilder.Mapping("IReadOnlyCollection<int>", "IEnumerable<int>", TestSourceBuilderOptions.WithDeepCloning);
-        TestHelper.GenerateMapper(source).Should().HaveSingleMethodBody("return global::System.Linq.Enumerable.ToArray(source);");
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+            var target = new int[source.Count];
```

Comment:
Wouldn't here a array clone be faster?

Reply:
`IReadOnlyCollection` can't use `Array.Clone`.

It did make me think about `ICollection.CopyTo`... I imagine in some cases using `ToArray` will be faster for synthetic mappings if the source implements `ICollection`.

For instance deep cloning `IReadOnlyCollection<int>` -> `int[]` with `ToArray` is probably faster than a `for` loop as the source will likely be `ICollection` or array and will use `Array.Clone` or `ICollection.CopyTo` internally.